src/software/new_geom/util/closest_point_test.cpp

```diff
@@ -46,3 +46,19 @@ TEST(ClosestPointTest, point_far_from_line)
     EXPECT_EQ(closestPointOnLine(p, l), expected);
     EXPECT_EQ(closestPointOnLine(l, p), expected);
 }
+
+TEST(GeomUtilTest, test_closest_lineseg_point)
+{
+    Segment seg(Point{-1, 1}, Point{1, 1});
+
+    EXPECT_TRUE((closestPointOnSeg(Point(0, 2), seg) - Point(0, 1)).length() < 0.00001);
+    EXPECT_TRUE((closestPointOnSeg(Point(-2, 1.5), seg) - Point(-1, 1)).length() <
+                0.00001);
+
+    seg = Segment(Point{-2, 1}, Point{1, 2});
+
+    EXPECT_TRUE((closestPointOnSeg(Point(1, 0), seg) - Point(0.4, 1.8)).length() <
+                0.00001);
+    EXPECT_TRUE((closestPointOnSeg(Point(-1.4, 1.2), seg) - Point(-1.4, 1.2)).length() <
+                0.00001);
```

Comment:
I think we want to use `equalWithinTolerance` for these tests, take a look at https://github.com/UBC-Thunderbots/Software/blob/master/src/software/test_util/test_util.h

Reply:
Done.